internal/core/src/segcore/SegmentInterface.cpp

```diff
@@ -62,35 +61,22 @@ SegmentInternalInterface::FillTargetEntry(const query::Plan* plan, SearchResult&
                        "Primary key field is not INT64 type");
             bulk_subscript(key_offset, results.ids_.data(), size, blob.data());
         }
-        blobs.emplace_back(std::move(blob));
-        element_sizeofs.push_back(sizeof(int64_t));
+        results.ids_data_ = std::move(blob);
```

Comment:
Just modify ids_data_ in previous bulk_subscript?

Reply:
fixed